src/app/beer_garden/local_plugins/manager.py

```diff
@@ -277,159 +343,225 @@ def stop_one(
 
     def _from_instance_id(self, instance_id: str) -> Optional[ProcessRunner]:
-        for runner in self._runners:
-            if runner.instance_id == instance_id:
-                return runner
+        for the_runner in self._runners:
+            if the_runner.instance_id == instance_id:
+                return the_runner
+
         return None
 
     def _from_runner_id(self, runner_id: str) -> Optional[ProcessRunner]:
-        for runner in self._runners:
-            if runner.runner_id == runner_id:
-                return runner
+        for the_runner in self._runners:
+            if the_runner.runner_id == runner_id:
+                return the_runner
+
         return None
 
-    def _restart(self, runner: ProcessRunner) -> ProcessRunner:
+    def _restart(self, the_runner: ProcessRunner) -> ProcessRunner:
         new_runner = ProcessRunner(
-            runner_id=runner.runner_id,
-            process_args=runner.process_args,
-            process_cwd=runner.process_cwd,
-            process_env=runner.process_env,
-            capture_streams=runner.capture_streams,
+            runner_id=the_runner.runner_id,
+            process_args=the_runner.process_args,
+            process_cwd=the_runner.process_cwd,
+            process_env=the_runner.process_env,
+            capture_streams=the_runner.capture_streams,
         )
 
-        self._runners.remove(runner)
+        self._runners.remove(the_runner)
         self._runners.append(new_runner)
 
         new_runner.start()
 
         return new_runner
 
-    def _stop_multiple(self, runners: Iterable[ProcessRunner] = None) -> None:
+    def _stop_multiple(
+        self, the_runners: Optional[Iterable[ProcessRunner]] = None
+    ) -> None:
         # If not specified, default to all runners
-        if runners is None:
-            runners = self._runners
+        if the_runners is None:
+            the_runners = self._runners
 
         # If empty, we're done
-        if len(runners) < 1:
+        if len(the_runners) == 0:
             return
 
-        shutdown_pool = ThreadPoolExecutor(len(runners))
+        shutdown_pool = ThreadPoolExecutor(len(the_runners))
         stop_futures: List[Future] = []
 
-        for runner in runners:
+        for the_runner in the_runners:
             stop_futures.append(
                 shutdown_pool.submit(
-                    self.stop_one, runner_id=runner.runner_id, send_sigterm=True
+                    self.stop_one, runner_id=the_runner.runner_id, send_sigterm=True
                 )
             )
 
         wait(stop_futures)
 
-        for runner in runners:
-            self._runners.remove(runner)
+        for the_runner in the_runners:
+            self._runners.remove(the_runner)
+
+    def _get_runner_id(self) -> str:
+        """Get a 10-letter string to serve as a runner ID."""
+        if self._runner_id_generator is None:
+            self._runner_id_generator = self._runner_id_generator_factory()
```

Comment:
I'm not the expert on generators, but isn't this going to effectively reset your stored `seen` set every time because you're calling the generator fresh each time?  Meaning I think you're going to end up with the same behavior, where duplicates are still very much possible.

It seems like we should just generate an actual uuid, but also I'm not sure you need to fix the issue you were trying to address here in this particular PR.

Reply:
Per our offline discussion, I completely glossed over the fact that you're' storing the `__runner_id_generator` and then only assigning if it isn't already set.  In which case disregard my concern and carry on.